staging/kos/README.md

```diff
@@ -188,6 +196,119 @@ remote NetworkAttachment node. The flows installed in that bridge will
+*X*, a conflict with *Y* is found, *Y*'s *VNI* and *CIDR* are compared to those
+stored in its *conflicts cache* entry. If they match, *X*'s status is updated
+by adding information on the conflict, and *X*'s namespaced name is added to
+the conflicts cache entry for *Y*. If they do not match, no further action is
+taken and *X* is processed again after a delay. When *Y* is deleted or its
+*VNI* and *CIDR* are updated, its *confilcts cache* entry is updated: the new
+*VNI* and *CIDR* override the old ones (in case of an update) and the list of
+conflicting subnets namespaced names is reset. All the namespaced names in the
+old list are enqueued so that their subnets can be re-validated: they did not
+pass validation because of a conflict with *Y*, but *Y* no longer exists or has
+changed (potentially making the conflict disappear), so they might be
+successfully validated now. If *X* was added unconditionally to *Y*'s conflicts
+cache, an update to *Y*'s *VNI* and *CIDR* which makes the conflict disappear
+might take place and be processed by the validator in between the live list
+against the API server and the addition of *X* to *Y*'s *conflicts cache*.
+This would mean *X* is added to *Y*'s conflicts cache so that it can be
+re-validated in case an update to *Y* makes the conflict disappear after such
+an update has actually happend and has already been processed. This means
+*X* will not be re-validated even if it is perfectly safe to use because it
+has no conflicts with other subnets. This problem is caused by a skew between
+*Y*'s version when its *conflicts cache* was last updated and the one returned
+by the extension API server. Thus, comparing the two versions (recorded in the
+form of *VNI* and *CIDR*) when adding *X* to *Y*'s *conflicts cache* allows
+to detect version skews and re-validate *X* after a delay, so that the skew
+has disappeared by the time *X* is processed again.
+
+One apparently odd design choice is having the validator retrieve all the
+subnets with a live list against the API server rather than a cache-based
+list using its Informer. The live list is worse performance-wise, but is
+necessary to avoid race conditions that can lead to two conflicting subnets
+having both `SubnetStatus.Validated` set to true. The subnets validator is a
+singleton, but Kubernetes makes no guarantee that there cannot be transients
+with more than one running. Assume two --- *V1* and *V2* --- are running and
+cache-based lists are used. Also assume a subnet *X* is created and shortly
+after another subnet *Y* in conflict with *X* is created. *X* and *Y* might
+appear in *V1* and *V2*'s Informers caches in opposite orders: *V1*
+processes *X* first, and when it does the cache-based list *Y* is not in
+the cache yet. Likewise, *V2* might validate *Y* without seeing *X*. Hence
+both *X* and *Y* are marked as validated even if they are in conflict. The
+problem is due to the fact that Informer's caches are not populated in order.
```

Comment:
the problem is due to the fact that an Informer does not necessarily deliver change notifications in order